app/backend/routes/venues.js

```diff
@@ -10,4 +13,59 @@ router.get('/', (req,res,next) => {
   res.send(output)
 })
 
+router.post('/verify',ensureAuthenticated, (req, res, next) => {
+  // console.log(req.body);
+  /* req.body
+  { 
+    name: 'Bitwise Industries',
+    business_key: 'SAMPLE KEY PHASE'  
+  }
+  */
+  let inputKey = req.body.key
+  let venue = venues().find(v => { return v.name === req.body.name; }) 
+
+  // TODO: I will also need to check if user has already entered code for the venue
+  if(venue.businessKey === inputKey) {
+    User.findById(req.user, function(err, user) {
+      
+      user.points += 1 //We will want to do something like venue.points if we implement different points for each venue
+      user.save(function(err) {
+        if(err) {
+          console.log("Error with saving the user", err)
+          res.send({text: "Sorry error with saving your points"})
+        }
+        console.log("Points added to", user.displayName)
+        res.send({ text: "Congrats! Here's a cookie.", points: user.points })
+      })
+    })
+  } else {
+    res.send({text: "Sorry key doesn't match."})
+  }
+})
+
 module.exports = router
+
+function ensureAuthenticated(req, res, next) {
```

Comment:
I would add this check into the users model in `../models/user.js` or as a `../services/user.js` and use the method as a service.